app/assets/javascripts/shopify_app/storage_access.js

```diff
@@ -0,0 +1,114 @@
+var ACCESS_GRANTED_STATUS = 'access granted';
+var ACCESS_DENIED_STATUS = 'access denied';
+
+function StorageAccessHelper(redirectInfo) {
+  this.redirectInfo = redirectInfo;
```

Comment:
Not crazy about the word `redirectInfo` as a name for a configuration-like object. Maybe `config` or `redirectData`.